src/FlightMap/Widgets/PhotoVideoControl.qml

```diff
@@ -101,14 +101,16 @@ Rectangle {
 
     function toggleShooting() {
         console.log("toggleShooting", _anyVideoStreamAvailable)
-        if (_mavlinkCamera && (_mavlinkCamera.capturesVideo || _mavlinkCamera.capturesPhotos) ) {
-            if(_mavlinkCameraInVideoMode) {
-                _mavlinkCamera.toggleVideo()
-            } else {
-                if(_mavlinkCameraInPhotoMode && !_mavlinkCameraPhotoCaptureIsIdle && _mavlinkCameraElapsedMode) {
-                    _mavlinkCamera.stopTakePhoto()
+        if (_mavlinkCamera) {
+            if (_mavlinkCamera.capturesVideo || _mavlinkCamera.capturesPhotos) {
+                if(_mavlinkCameraInVideoMode) {
+                    _mavlinkCamera.toggleVideo()
                 } else {
-                    _mavlinkCamera.takePhoto()
+                    if(_mavlinkCameraInPhotoMode && !_mavlinkCameraPhotoCaptureIsIdle && _mavlinkCameraElapsedMode) {
+                        _mavlinkCamera.stopTakePhoto()
+                    } else {
+                        _mavlinkCamera.takePhoto()
+                    }
                 }
             }
         } else if (_onlySimpleCameraAvailable || (_simpleCameraAvailable && _anyVideoStreamAvailable && _videoStreamInPhotoMode && !videoGrabRadio.checked)) {
```

Comment:
Hmmm. The above changes aren't going to fall through to the other elses the same way they used to. To avoid this I would do the following with a comment to explain with it's so crazy:
```
// This whole mavlinkCameraCaptureVideoOrPhotos stuff is to work around some strange qml boolean testing // behavior which wasn't working correctly. This should work:
//    if (_mavlinkCamera && (_mavlinkCamera.capturesVideo || _mavlinkCamera.capturesPhotos) ) {
// but it doesn't for some strange reason. Hence all the stuff below...

var mavlinkCameraCaptureVideoOrPhotos = false
if (_mavlinkCamera) {
    if (_mavlinkCamera.capturesVideo || _mavlinkCamera.capturesPhotos) {
        mavlinkCameraCaptureVideoOrPhotos = true
    }
}
if (mavlinkCameraCaptureVideoOrPhotos) {
} else if (...)
```

Reply:
I like this idea. Pushed up changes.

Still super weird that qml is behaving this way. I wonder if this is a bug in 5.15.2 or just something that is happening in this particular area.